thrust/detail/vector_base.inl

```diff
@@ -87,6 +96,20 @@ template<typename T, typename Alloc>
   return *this;
 } // end vector_base::operator=()
 
+#if __cplusplus >= 201103L
+  template<typename T, typename Alloc>
+    vector_base<T,Alloc> &
+      vector_base<T,Alloc>
+        ::operator=(vector_base &&v)
+  {
+    //We don't check for self move assignement : undefined behaviour
```

Comment:
To avoid undefined behaviour at runtime, would it  be rather better to `assert(this != &v)`.